java/statestore/src/main/java/sleeper/statestore/transactionlog/DynamoDBTransactionLogStore.java

```diff
@@ -0,0 +1,109 @@
+    private static final String BODY = "BODY";
+
+    private final InstanceProperties instanceProperties;
+    private final TableProperties tableProperties;
+    private final AmazonDynamoDB dynamo;
+    private final TransactionSerDe serDe;
+
+    DynamoDBTransactionLogStore(
+            InstanceProperties instanceProperties, TableProperties tableProperties, AmazonDynamoDB dynamo) {
+        this.instanceProperties = instanceProperties;
+        this.tableProperties = tableProperties;
+        this.dynamo = dynamo;
+        this.serDe = new TransactionSerDe(tableProperties.getSchema());
+    }
+
+    @Override
+    public void addTransaction(StateStoreTransaction transaction, long transactionNumber) {
+        dynamo.putItem(new PutItemRequest()
+                .withTableName(instanceProperties.get(TRANSACTION_LOG_TABLENAME))
+                .withItem(new DynamoDBRecordBuilder()
+                        .string(TABLE_ID, tableProperties.get(TableProperty.TABLE_ID))
+                        .number(TRANSACTION_NUMBER, transactionNumber)
+                        .string(TYPE, transaction.getClass().getName())
+                        .string(BODY, serDe.toJson(transaction))
+                        .build())
+                .withConditionExpression("attribute_not_exists(#Number)")
+                .withExpressionAttributeNames(Map.of("#Number", TRANSACTION_NUMBER)));
+    }
+
+    @Override
+    public Stream<StateStoreTransaction> readTransactionsAfter(long lastTransactionNumber) {
+        return streamPagedItems(dynamo, new QueryRequest()
+                .withTableName(instanceProperties.get(TRANSACTION_LOG_TABLENAME))
+                .withConsistentRead(true)
+                .withKeyConditionExpression("#TableId = :table_id AND #Number > :number")
+                .withExpressionAttributeNames(Map.of("#TableId", TABLE_ID, "#Number", TRANSACTION_NUMBER))
+                .withExpressionAttributeValues(new DynamoDBRecordBuilder()
+                        .string(":table_id", tableProperties.get(TableProperty.TABLE_ID))
+                        .number(":number", lastTransactionNumber)
+                        .build())
+                .withReturnConsumedCapacity(ReturnConsumedCapacity.TOTAL))
+                .flatMap(item -> readTransaction(item).stream());
+    }
+
+    private Optional<StateStoreTransaction> readTransaction(Map<String, AttributeValue> item) {
+        String className = item.get(TYPE).getS();
+        try {
+            Class<?> type = Class.forName(className);
+            if (!StateStoreTransaction.class.isAssignableFrom(type)) {
+                LOGGER.warn("Found non-transaction type for table {} transaction {}: {}",
+                        item.get(TABLE_ID).getS(), item.get(TRANSACTION_NUMBER).getS(), className);
+                return Optional.empty();
+            }
+            Class<? extends StateStoreTransaction> transactionType = (Class<? extends StateStoreTransaction>) type;
```

Comment:
I'd like to add type strings for each transaction type, so that if we changed the class names or package names this would still work.